fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -52,28 +108,81 @@ class SignalSparsity:
             conv.weight.data = 3d_sparser.get_sst_dst_weight(conv.weight.data)
     """
 
-    def __init__(self) -> None:
-        pass
+    def __init__(
+        self,
+        algo: Algo = Algo.FFT,
+        sst_top_k_dim: Optional[int] = -1,
+        sst_top_k_element: Optional[int] = None,
+        sst_top_k_percent: Optional[float] = None,
+        dst_top_k_dim: Optional[int] = -1,
+        dst_top_k_element: Optional[int] = None,
+        dst_top_k_percent: Optional[float] = None,
+    ) -> None:
+
+        self._sst_top_k_dim = sst_top_k_dim
+        self._sst_top_k_element = sst_top_k_element
+        self._sst_top_k_percent = sst_top_k_percent
+        self._dst_top_k_dim = dst_top_k_dim
+        self._dst_top_k_element = dst_top_k_element
+        self._dst_top_k_percent = dst_top_k_percent
+
         self._validate_conf()
+        self._transform = torch.fft.fft if algo == Algo.FFT else self._dct_transform  # type: ignore
 
     def _validate_conf(self) -> None:
         """Validating the config is valid.
 
-        For example, not both top_k_element and top_k_percent is set.
+        This includes asserting the following:
+        1. validating that both top_k_element and top_k_percent is not set.
+        2. Asserting that if top_k_percent is not None, it is a valid number for percentage.
 
-        this should assert fail if checking fails.
+        This asserts fail if the checking fails.
         """
-        pass
+        # assert that both top_k_elements and top_k_percent aren't set for sst and dst
+        assert (self._sst_top_k_element is None) ^ (
+            self._sst_top_k_percent is None
+        ), "One and only one of sst_top_k_element and sst_top_k_percent has to be provided as argument"
+        assert (self._dst_top_k_element is None) ^ (
+            self._dst_top_k_percent is None
+        ), "One and only one of dst_top_k_element and dst_top_k_percent has to be provided as argument"
+
+        # assert, if top_k_percent is not None, it is a valid number for percentage.
+        assert self._sst_top_k_percent is None or (
+            0.0 < self._sst_top_k_percent <= 100.0
+        ), "The value for the argument sst_top_k_percent has to be in the interval (0, 100]"
+
+        assert self._dst_top_k_percent is None or (
+            0.0 < self._dst_top_k_percent <= 100.0
+        ), "The value for the argument dst_top_k_percent has to be in the interval (0, 100]"
```

Comment:
think how to simplified these checks. make it shorter and easy to understand at the same time.

Reply:
at least same logic on sst/dst should not be repeated.

Reply:
DRY principle: don't repeat yourself.